planner/core/exhaust_physical_plans.go

```diff
@@ -588,46 +588,86 @@ func (p *LogicalJoin) buildIndexJoinInner2TableScan(
 	if tblPath == nil {
 		return nil
 	}
-	pkCol := ds.getPKIsHandleCol()
-	if pkCol == nil {
-		return nil
-	}
 	keyOff2IdxOff := make([]int, len(innerJoinKeys))
 	newOuterJoinKeys := make([]*expression.Column, 0)
-	pkMatched := false
-	for i, key := range innerJoinKeys {
-		if !key.Equal(nil, pkCol) {
+	var ranges []*ranger.Range
```

Comment:
Can we extract line 593~line 619 to a function, and reuse it with `buildIndexJoinInner2IndexScan`